sdk/core/AzureCore/Source/Util/ConvertingInitializers.swift

```diff
@@ -57,23 +57,64 @@ extension Date {
             formatter.dateFormat = "EEE, dd MMM yyyy HH:mm:ss z"
             return formatter
         }()
+
+        static var iso8601: DateFormatter = {
+            let formatter = DateFormatter()
+            formatter.calendar = Calendar(identifier: .iso8601)
+            formatter.locale = Locale(identifier: "en_US_POSIX")
+            formatter.timeZone = TimeZone(abbreviation: "GMT")
+            formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZZZZZ"
+            return formatter
+        }()
+
+        static var iso8601WithMilliseconds: DateFormatter = {
+            let formatter = DateFormatter()
+            formatter.calendar = Calendar(identifier: .iso8601)
+            formatter.locale = Locale(identifier: "en_US_POSIX")
+            formatter.timeZone = TimeZone(abbreviation: "GMT")
+            formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZZZZZ"
+            return formatter
+        }()
     }
 
     public enum Format {
         case rfc1123
+        case iso8601
```

Comment:
Why are you not exposing `iso8601WithMilliseconds` here? Is there a way to specify the `WithMilliseconds` aspect with associated values?